components/Ripple/src/MDCRippleTouchController.h

```diff
@@ -0,0 +1,184 @@
+
+/** Delegate to extend the behavior of the touch control. */
+
+/**
+ A delegate to extend the behavior of the touch controller.
+ */
+@property(nonatomic, weak, nullable) id<MDCRippleTouchControllerDelegate> delegate;
+
+/** Gesture recognizer used to bind touch events to ripple. */
+
+/**
+ The gesture recognizer used to bind the touch events to the ripple.
+ */
+@property(nonatomic, strong, readonly, nonnull) UILongPressGestureRecognizer *gestureRecognizer;
+
+/**
+ The selection gesture recognizer used to bind the touch events related to selection to the ripple.
+ */
+@property(nonatomic, strong, readonly, nullable)
+    UILongPressGestureRecognizer *selectionGestureRecognizer;
+
+/**
+ This BOOL tells the touch controller if to allow selection and all the logic and visuals
+ that come with it, for this ripple.
+
+ Defaults to NO.
+ */
+@property(nonatomic) BOOL allowsSelection;
+
+/**
+ This BOOL is set to YES if the ripple is currently selected, or NO otherwise.
+ It only has significance if selectionMode is activated.
+
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isSelected) BOOL selected;
+
+/**
+ This BOOL is set to YES if the ripple is currently in the selection mode, or NO otherwise.
+
+ Defaults to NO.
+ */
+@property(nonatomic) BOOL selectionMode;
+
+/**
+ Sets the color of the ripple for state.
+
+ @param rippleColor The ripple color to set the ripple to.
+ @param state The state of the ripple in which to set the ripple color.
+ */
+- (void)setRippleColor:(nullable UIColor *)rippleColor forState:(MDCRippleState)state;
+
+/**
+ Gets the ripple color for the given state.
+
+ @param state The ripple's state.
+ @return the color of the ripple for state.
+ */
+- (nullable UIColor *)rippleColorForState:(MDCRippleState)state;
+
+/**
+ Sets the alpha of the ripple for state.
+
+ @param rippleAlpha The ripple alpha to set the ripple to.
+ @param state The state of the ripple in which to set the ripple alpha.
+ */
+- (void)setRippleAlpha:(CGFloat)rippleAlpha forState:(MDCRippleState)state;
+
+/**
+ Gets the ripple alpha for the given state.
+
+ @param state The ripple's state.
+ @return the alpha of the ripple for state.
+ */
+- (CGFloat)rippleAlphaForState:(MDCRippleState)state;
+
+/**
+ The current state of the ripple.
+ */
+@property(nonatomic, readonly) MDCRippleState state;
+
+/** Unavailable, please use initWithView: instead. */
+
+/**
+ Unavailable, please use `initWithView` instead.
+ */
+- (nonnull instancetype)init NS_UNAVAILABLE;
+
+/**
+ Initializes the controller.
+
+ @param view The view that responds to the touch events for the ripple, and the view which the
+ ripple is added as a subview to.
+ */
+- (nonnull instancetype)initWithView:(nonnull UIView *)view NS_DESIGNATED_INITIALIZER;
+
+@end
+
+/**
+ Delegate methods for MDCRippleTouchController.
+ */
+@protocol MDCRippleTouchControllerDelegate <NSObject>
+@optional
+
+/**
+ Controls whether the ripple touch controller should be processing touches.
```

Comment:
Is this worded how you intended? "Controls whether the ripple touch controller should process touches" as a possible alternative.